File: 02-arrays/easy/09-maxOneCount.cpp
// Problem Statement: Given an array that contains only 1 and 0 return the count of maximum
// consecutive ones in the array..
#include <bits/stdc++.h>
using namespace std;
int findMaxConsecutiveOnes(vector<int> &arr) {
    int maxi = 0;
    int cnt = 0;
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] == 1)
            cnt++;
        else
            cnt = 0;
        maxi = max(maxi, cnt);
    }
    return maxi;
}

File: 02-arrays/easy/09-maxOneCount.go
package main

import "fmt"

func MaxOneCount(arr []int) int {
	maxi := 0
	count := 0
	for _, ele := range arr {
		if ele == 1 {
			count++
		} else {
			count = 0
		}
		if count > maxi {
			maxi = count
		}
	}
	return maxi
}

func main() {
	arr := []int{0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1}
	res := MaxOneCount(arr)
	fmt.Println(res)
}
